Support the "Xcode (iOS)" Projucer exporter

McMartin opened the pull request:
Related to #458.

This PR is made of the following commit "groups":
- commit 7fe3eafe29e464699327d779c3bdd2a9d7e1f113 documents per-exporter CMake requirements
- commits 6828d9b3ef3ad97f86d0c0ad22a805a00a03ab95 and 940cf8b1036b2b82b07d344493bf0a63684257bf only deal with renaming
- commits from e9fb724dcb7c6419e06f28df710eb3d5f0e7044c to a742a5def03ad56acc667bb4b36b4781868e7917 add the core of the iOS support
- commits from 887ade9ea5fec783753e0f23de695766a5fd31f6 to ee2314e241467496e312f03f9ea4b419d672c120 update generated `CMakeLists.txt` files (most of the changes were done by `Jucer2Reprojucer`)
- commits from 50919a512ce99f91cedaadb7eb2fe038be505c62 to 4e181866521bc1e5cb3058e4de84fa6c64128f26 add support for various settings that are specific to the "Xcode (iOS)" exporter
- commits 9e786c3a1bd6af7709450e9cd31a189681de5b0e and dc6bc364b37732d898492c401b9da29ea334d8d0 extend CI and polish the README

@MartyLake: please review, thanks!

MartyLake commented on cmake/XcassetsBuilder/main.cpp:
```diff
@@ -0,0 +1,71 @@
+  const auto smallIconImageFile =
+    args.at(2) == "<None>"
+      ? File{}
+      : File::getCurrentWorkingDirectory().getChildFile(juce::String{args.at(2)});
```
Will this also work if path is absolute ?

McMartin replied:
It will indeed work. From https://github.com/WeAreROLI/JUCE/blob/5.4.5/modules/juce_core/files/juce_File.h#L243-L257:
```cpp
        If the string is actually an absolute path, it will be treated as such, e.g.
            File ("/moose/fish").getChildFile ("/foo.txt") will produce "/foo.txt"
```

MartyLake commented on cmake/Reprojucer.cmake:
```diff
@@ -1938,6 +1988,12 @@ function(jucer_project_end)
 
+  if(JUCER_IOS)
+    _FRUT_create_xcassets_folder_from_icons(JUCER_XCASSETS)
+
```
empty line leftover ?

McMartin replied:
Addressed in https://github.com/McMartin/FRUT/pull/551/commits/74dc8d7e4588683dad9cc1c60d642d14d51f2d94#diff-d7ac1cfde9bf2d7ef9cccf6fab14bc6eR1990-R1991.

MartyLake commented:
LGTM, even if it's hard to say I read every single line of it

McMartin commented:
[First-class support for iOS was added in CMake 3.14](https://cmake.org/cmake/help/latest/release/3.14.html#platforms). When I opened this PR, I didn't want to make that a hard requirement, since the code of FRUT itself only requires CMake version 3.4 minimum. However, after a bit of testing on macOS, and considering that Visual Studio 2019 is also only supported since CMake 3.14, I changed my mind. In this new version of the PR, CMake 3.14 is required for iOS. Instead of calling `cmake -G Xcode -DJUCER_IOS=ON` and using `JUCER_IOS` in `Reprojucer.cmake`, we now call `cmake -G Xcode -DCMAKE_SYSTEM_NAME=iOS` and we use [`IOS`](https://cmake.org/cmake/help/v3.14/variable/IOS.html) in `Reprojucer.cmake`.

@MartyLake: please re-review this PR. To help you understand what changed, you can read the [diff between bb0cb70 and 96933d1](https://github.com/McMartin/FRUT/compare/bb0cb703b8ac1782718991cd38b64ccc0401fc1b..96933d1f15cb2fd787c9d36d4e79c4ce5c20360d). I also updated the PR description with the new commit groups. Thanks!

McMartin commented:
We hit a new problem. `Reprojucer.cmake` needs to build its helper executables (`BinaryDataBuilder`, `IconBuilder`, `PListMerger`, and the new `XcassetsBuilder`), but these executables can only be built and run on macOS, not iOS. Since `Reprojucer.cmake` uses the `try_compile` CMake command, `CMAKE_SYSTEM_NAME=iOS` (and everything that comes with it) is passed down when configuring the helper executables. This obviously fails because the Cocoa framework cannot be found in the `iphoneos` SDK.

@MartyLake: please don't review yet, fixing this problem might require quite some changes.

McMartin commented:
After reading more about `try_compile` I came to the conclusion that it is not the right tool in the first place. `try_compile` should be used to determine if something can/would compile, it shouldn't be used to just build and install other CMake projects at configure time. I'll open a new PR to replace `try_compile` by something else in `_FRUT_build_and_install_helper_exe`, and then I'll rebase this one again. Stay tuned!

McMartin commented:
@MartyLake: I've rebased the branch on top of the latest `master` after merging #553 and force-pushed. This PR is now ready for review again, please have another look, thanks!

MartyLake commented:
LGTM, no obvious typo 🗡
